MAUI/DataForm/localization.md

```diff
@@ -0,0 +1,182 @@
+
+#### Using attribute
+
+The `ValidationAttribute` error message can be localized using [ErrorMessageResourceType](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) and [ErrorMessageResourceName](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) properties which are used to get a localized error messages from Localization Resource File (.Resx) based on culture.
+
+{% tabs %}
+{% highlight C# %}
+
+[Required(ErrorMessage = "Value should not be empty", ErrorMessageResourceName = "ErrorMessage", ErrorMessageResourceType = typeof(DataFormLocalization))]
+[StringLength(15, ErrorMessage = "Enter proper name", ErrorMessageResourceName ="ErrorMessageLength", ErrorMessageResourceType =typeof(DataFormLocalization))]
+public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can also localize the data form error message and valid message in the  `ValidateProperty` event of `SfDataForm` by using the Resources (.Resx) file.
+
+{% tabs %}
+{% highlight C# %}
+
+[DataFormDisplayOptions(ValidMessage = "Text length is enough")]
+[Required(ErrorMessage = "Enter proper name")]
+[StringLength(15, ErrorMessage = "Enter proper name")]
+public string Name { get; set; }
+
+this.dataForm.ValidateProperty += DataForm_ValidateProperty;
+
+private void DataForm_ValidateProperty(object sender, DataFormValidatePropertyEventArgs e)
+{
+    e.ErrorMessage = DataFormLocalization.ErrorMessage;
+    e.ValidMessage = DataFormLocalization.ValidMessage;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Localizing data form picker editors
+
+You can localize DataForm list items (Picker, AutoComplete, RadioGroup, ComoboBox) `ItemsSource` using ResourceType property of [Display](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute?view=net-7.0) attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(ResourceType = typeof(DataFormLocalization))]
+public string Gender { get; set; }
+
+this.dataForm.RegisterEditor("Gender", DataFormEditorType.RadioGroup);
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem?.FieldName == "Gender")
+    {
+        var list = new List<string>();
+        list.Add("Male");
+        list.Add("Female");
+        list.Add("Other");
+        (e.DataFormItem as DataFormRadioGroupItem).ItemsSource = list;            
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+Here, the radio group items source get localized based on culture from Localization Resource File (.Resx).
```

Comment:
Here, the radio group items source gets localized based on culture from Localization Resource File (.Resx).

Reply:
content changed